components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesBookmarksStorage.kt

```diff
@@ -95,8 +95,9 @@ open class PlacesBookmarksStorage(context: Context) : PlacesStorage(context), Bo
             } else {
                 0
             }
-            reader.getRecentBookmarks(limit).filter { it.dateAdded >= threshold }
+            reader.getRecentBookmarks(limit)
                 .map { it.asBookmarkNode() }
+                .filter { it.dateAdded >= threshold }
```

Comment:
[getRecentBookmarks](https://github.com/mozilla/application-services/blob/ee10746170b4028d0b686dc52d6d7fbee24b59ee/components/places/android/src/main/java/mozilla/appservices/places/Bookmarks.kt#L121) now returns a [BookmarkItem](https://github.com/mozilla/application-services/pull/4770/files#diff-02c905e5e17812b206a3dcd667061c0b4420847f930fe3f455e89f9246c9e6acR362-R367) instead of a BookmarkNode due to how the API churned and to consolidate some of types -- which means I had to flip them due to needing to make it a bookmarknode before using the filter otherwise a-c doesn't build for me

Reply:
Okay, cool. Let's let this ride as-is in the PR (any changes w/ the toolchain task involved are extra annoying), maybe we can flip it around afterwards.